validator/walk.go

```diff
@@ -197,6 +204,11 @@ func (w *Walker) walkSelection(parentDef *gqlparser.Definition, it gqlparser.Sel
 			v(w, parentDef, def, &it)
 		}
 
+		if w.Observers.visitedFrags[it.Name] {
```

Comment:
whats clearing this? 
my understanding is this should be valid:
```graphql
{
    frag AFrag on A {
        bone { ... BFrag }
        btwo { ... BFrag }
    }
    Frag BFrag on B {
        name
    }
}

```

Reply:
agreed.
This test case raise `fatal error: stack overflow` error.
https://github.com/vektah/gqlparser/blob/b806ed27eeba6ddc72f07e4fd26acd6aa8a3b5c6/spec/validation/NoFragmentCycles.yml#L55-L65
I thought that processing should be canceled here, but is it wrong?

Reply:
This code block can't prevent inifinite loop at [this test case](https://github.com/vektah/gqlparser/blob/b806ed27eeba6ddc72f07e4fd26acd6aa8a3b5c6/spec/validation/NoUndefinedVariables.yml#L101-L115) 🤔